### PR TITLE
Adding package tool

### PR DESCRIPTION
**Pull request by dvalinrh**

### CODE REVIEW

File: package_tool [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		--is_installed)
+			is_installed=$2
+                        shift 2
+		;;
+		--no_packages)
+			if [[ $2 == "1" ]]; then
+				exit 0
+			fi
+                        shift 2
+		;;
```
Comment (kdvalin) [inline package_tool:84]: For this case, wouldn't it make sense to have `--no_packages` be a zero argument flag?

Simply the existence of `--no_packages` would indicate to skip install

File: package_tool [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+else
+	if [[ -f "/bin/dnf" ]]; then
+		install_cmd="/bin/dnf"
+	elif [[ -f "/bin/yum" ]]; then
+		install_cmd="/bin/yum"
+	fi
+	if [[ $install_cmd == "" ]]; then
+		exit_out "package_install: Do not know what to use to install packages with" 1
+	fi
+fi
```
Comment (kdvalin) [inline package_tool:129]: Convert this to a switch to clearly establish the flow of the script.  Also allows for easy expansion later

File: package_tool [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+if [[ $is_installed != "" ]]; then
+	yum list installed | grep -q $is_installed
```
Comment (kdvalin) [inline package_tool:132]: Should be generalized for Ubuntu/Fedora/Whatever

**Review by kdvalin — CHANGES_REQUESTED** [review]
Good first jab, might want to separate out logic based on the package manager.

So for instance we can have `apt_update` `dnf_update` and `yum update` functions that can be called with `${install_cmd}_update`

File: package_tool (hunk 1 above)
Comment (dvalinrh) [reply package_tool:84]: Trying to push the logic from the wrapper into this.  At the wrapper level, it is a variable, and we pass that variable as part of --no_packages.  End goal is for the wrapper to simply say I want this package, and everything is done in this script.

File: package_tool (hunk 3 above)
Comment (dvalinrh) [reply package_tool:132]: Oops, missed one. Fixed.

**Review by kdvalin — APPROVED** [review]
LGTM for now

**Pull request merged**